### DISCUSSION TITLE
Doxygen documentation binary image files on github pages

### DISCUSSION THREAD

**Discussion by BYUignite** [community/community#23705] [Other Feature Feedback, Questions, & Ideas]
<p>I would like to use Github pages in conjunction with Doxygen for a C++ project. My Doxygen documentation can be fully generated without inclusion of the files in the repository. However, my understanding is that these generated files need to be part of/tracked by the repository in order to be included in the github pages. This is fine for the text files, but what about the image files (png) that are generated. I don’t want the image files to be part of the repository if those binary files are going to change every time we regenerate the documentation. Is it possible for github pages to use content that is outside the repository, like, generated by Doxygen, but not under version control? Alternatively, if the repository only ever kept the latest version of image files, that could work, but that seems contrary to how git works…</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/byuignite">@BYUignite</a>, welcome to the GitHub Support Community!</p>
<p>Most of the time if you want to keep your Pages site completely separate from your repository then we’d recommend storing it on the <code>gh-pages</code> branch. This keeps your site’s content out of the way of the rest of your repository.</p>
<p>Alternatively, if you want to keep your site in line with the history of your project, you can use the <code>docs</code> folder to store your site’s content. However, using this folder means you’ll need to include image assets inside the main branch of your repository, so if there is a large number of them then this may impede clone time/increase your repo’s size.</p>
<p>GitHub Pages <em>can</em> use externally hosted image files, so if you want to host your image files on an external server then you can do so. Of course you’ll need to manage the upload of these assets separately, but pointing to external URLs for these image assets will work. We don’t have any direct integrations that do this though so you’ll need to deal with managing that part yourself.</p>

**Comment by BYUignite** [comment]
<p>Thank you for the prompt reply. This is helpful. I will look into the gh-pages branch and consider the other options you suggest. --David</p>
